ethstats/ethstats.go

```diff
@@ -211,93 +253,215 @@ func (s *Service) loop() {
 				case txCh <- struct{}{}:
 				default:
 				}
-
 			// node stopped
 			case <-txSub.Err():
 				break HandleLoop
 			case <-headSub.Err():
 				break HandleLoop
+			case delegateSignMsg := <-istDelegateSignCh:
+				var statsPayload StatsPayload
+				err := json.Unmarshal(delegateSignMsg.Payload, &statsPayload)
+				if err == nil {
+					if s.backend.ProxyNode() != nil {
+						select {
+						case signCh <- statsPayload:
+						default:
+						}
+					}
+					if s.backend.ProxiedPeer() != nil {
+						select {
+						case sendCh <- statsPayload:
+						default:
+						}
+					}
+				}
 			}
 		}
 		close(quitCh)
 	}()
+
 	// Loop reporting until termination
 	for {
-		// Resolve the URL, defaulting to TLS, but falling back to none too
-		path := fmt.Sprintf("%s/api", s.host)
-		urls := []string{path}
+		if s.backend.ProxyNode() != nil {
```

Comment:
I find myself getting confused what this is saying, maybe a comment saying "if this is a proxied validator"